x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/transactions/transactions_groups_alerts.spec.ts

```diff
@@ -111,6 +108,7 @@ export default function ApiTest({ getService }: DeploymentAgnosticFtrProviderCon
       before(async () => {
         roleAuthc = await samlAuth.createM2mApiKeyWithRoleScope('admin');
         apmSynthtraceEsClient = await synthtrace.createApmSynthtraceEsClient();
+        await apmSynthtraceEsClient.clean();
```

Comment:
Shouldn't it be done in the `after` hook of the previous test suite? I'm afraid that the issue with other suite not properly cleaning after its run.

Reply:
It's happening on `after` as well, this is just to force a cleanup before starting the test as it seems in MKI some information is not being properly removed and this test was flaky

Reply:
is the intention here to clean `.alert` indices? If so, `clean` will just remove docs from these indices: https://github.com/elastic/kibana/blob/main/packages/kbn-apm-synthtrace/src/lib/apm/client/apm_synthtrace_es_client/index.ts#L39

Reply:
the intention is to clean what we add manually with synthtrace + all the alerts, as for some tests we add [transactions manually](https://github.com/elastic/kibana/pull/201755/files#diff-b13bc13f36ef42b3f73eba8ffe9372e3d899bbef6888823103d953c5f5cbbb4eR121) and they don't get removed with the cleanUpAlerts, as they are transactions and not alerts.

Reply:
and the apm indices are also causing the flakiness?

Reply:
I think they may be the source of the problem, something gets added by synthtrace in some tests, the cleanup function doesn't wait or this test executes before cleaning everything, and we end up with inconsistent data.

As I said, I haven't been able to test this locally, as in local it's always green. It only happened on CI / Automated runs

Reply:
Makes sense.

If you are curious to try to find the root cause and want to spend some (more) time on it, I suspect the issue originates in the [`latency.spec.ts`](x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/transactions/latency.spec.ts) test. It runs before this one and creates transactions with high latency. I don't know why the test is not cleaning up. 

We've seen synthtrace promises misbehaving on tests, it could be that synthtrace is still creating transactions even after the `before` finishes. What usually solves it is replacing `await` with `return` [here](https://github.com/crespocarlos/kibana/blob/main/x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/transactions/latency.spec.ts#L89) and let mocha handle the promise resolution.

What you could also do is run `after(() => apmSynthtraceEsClient.clean());`  in the parent `description` to ensure the data is cleaned.

Up to you though.

Reply:
That's so useful, will take a look and see if I can fix it! Thanks

Reply:
applied the changes in https://github.com/elastic/kibana/pull/201755/commits/73e7d280884c5ad931b82fbb2593e11d38a14797